src/modules/caen/CaenBase.cpp

```diff
@@ -231,24 +223,34 @@ void CaenBase::processingThread() {
       usleep(10);
     }
 
-    if (ProduceTimer.timeout()) {
+    if (ProduceTimer.timeout()){
       // XTRACE(DATA, DEB, "Serializer timer timed out, producing message now");
       RuntimeStatusMask =
           RtStat.getRuntimeStatusMask({ITCounters.RxPackets, Counters.Events,
                                        Counters.KafkaStats.produce_bytes_ok});
+      // Produce messages if there are events
+      for (auto &Serializer: Serializers) {
+        Serializer->produce();
+      }
+      MonitorSerializer->produce();
```

Comment:
Is this produce call necessary, since the same method is called every time the `MonitorSerializer` gets data (line 211 above)?

Reply:
That indeed looks like it is unnecessary